### ISSUE TITLE
Wrong images matched?

### ISSUE THREAD

**Issue by kapitanluffy** [vitali-fedulov/similar.pictures#3]
I haven't fully read how it is implemented but I assume these images shouldn't be similar? 

![image](https://github.com/vitali-fedulov/similar.pictures/assets/485799/abbb2bd1-a041-4f76-8d33-424a7d84ce6c)

It is just a small chunk of a single cluster. That cluster treats 1k+ images as similar

**Comment by vitali-fedulov** [comment]
I suspect these images have identical but incorrect EXIF thumbnails. To see the thumbnails, use the Jpeg thumbnail reader in Tools section of the web-site.

Try to run it with disabled box "Option to accelerate the search" and let me know.

**Comment by kapitanluffy** [comment]
I tried it and it does not show some images

**Comment by vitali-fedulov** [comment]
Jpeg thumbnail reader?

**Comment by kapitanluffy** [comment]
> Jpeg thumbnail reader?

Yep

**Comment by vitali-fedulov** [comment]
Correct. Jpeg thumbnail reader uses a very simple EXIF reading algorithm. For some images it does not work. I am not planning to fix it, as it is not commonly used, if at all.

Have you tried clustering with disabled box "Option to accelerate the search"? Did it work correctly?

**Comment by kapitanluffy** [comment]
Yes I tried that also. I think I know the problem now. Maybe because there's an all-black image. Here's a small sample of images that were grouped as similar

[similar_pictures_test.zip](https://github.com/vitali-fedulov/similar.pictures/files/15329022/similar_pictures_test.zip)
